kotlin-native/Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/Indexer.kt

```diff
@@ -267,8 +260,11 @@ internal class NativeIndexImpl(val library: NativeLibrary, val verbose: Boolean
                     }
                     fieldCursor.type.kind == CXType_Record -> {
                         // TODO: clang_Cursor_getOffsetOfField is OK for anonymous, but only for the 1st level of such nesting
+                        val declCursor = clang_getTypeDeclaration(fieldCursor.type)
                         AnonymousInnerRecord(
-                                convertCursorType(fieldCursor) as RecordType,
+                                RecordType(createStructDecl(declCursor).also {
```

Comment:
This looks better, but still `StructDecl` (and `RecordType`) seems kinda useless here, and it should be pretty easy to get rid of.

`findNamedParent` still looks redundant: why do we need to _find_ parent, if we already _know_ it? I mean, for the root struct it is the struct itself, and for nested anonymous structs we know the exact parent here.
Alternatively, we could also make parents unnecessary, if we stopped using the root struct as the base for field offsets (which also makes some sense). But I believe this might depend on the way other issues are handled.